File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int current = 0;
        int max = nums.length;
        boolean foundSolution = false;
        int[] solution = new int[2];
        
        while (!foundSolution && current < max) {
            for (int i = 0; i < nums.length; i++) {
                if ((nums[current] + nums[i] == target) && current != i) {
                    foundSolution = true;
                    solution[0] = current;
                    solution[1] = i;
                }
            }
            current++;
        }
        return solution;
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        # {} creates an empty dictionary. 
        # A dictionary stores key-value pairs
        # kind of like a real dictionary maps words to definitions.
        seen = {}

        # The i, num part is called unpacking, enumerate returns a pair (index, value)
        for i, num in enumerate(nums):
            complement = target - num

            if complement in seen:
                return [seen[complement], i]

            seen[num] = i
        